### PR TITLE
#170947501 super admin can assign roles

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
- add a feature whereby a super admin can assign roles

#### Description of Task to be completed?
- added component for the feature
- added actions and reducers for the feature
- added storybook stories for the feature
- added tests for the feature

#### How should this be manually tested?
- clone repo
- run `npm run start:dev`
- use the following URL: `http://localhost:3000/login` to login as super admin
- use the following URL: `http://localhost:3000/dashboard`

#### What are the relevant pivotal tracker stories?
- [#170947501](https://www.pivotaltracker.com/story/show/170947501)

#### Screenshots (if appropriate)
super admin

supllier

#### Questions:

### CODE REVIEW

**Review by izzett222 — CHANGES_REQUESTED** [review]
hey @Baraka-Mugisha could you make sure that when a role is changed, it is reflected on the table rather than having to reload the page to see the changes

**Comment by bbaime98** [comment]
image
- Shouldn't I see the option of assigning a manager after clicking on the assign manager button?

**Comment by Baraka-Mugisha** [comment]
> hey @Baraka-Mugisha could you make sure that when a role is changed, it is reflected on the table rather than having to reload the page to see the changes

The component is now updated!!

**Comment by Baraka-Mugisha** [comment]
> image
> 
> * Shouldn't I see the option of assigning a manager after clicking on the assign manager button?

I believe that is another feature.

**Comment by bbaime98** [comment]
> > image
> > 
> > * Shouldn't I see the option of assigning a manager after clicking on the assign manager button?
> 
> I believe that is another feature.

oh sure, it is.

File: src/components/dashboards/admin.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    this.updateUsers = this.updateUsers.bind(this);
+  }
+
+  async componentDidMount() {
+    const actionOutput = await this.props.getUsersAction();
+    this.setState({ users: actionOutput.payload.data });
+  }
+
+  async componentDidUpdate() {
+    console.log('.............................ADMIN UPDATED (admin.js)........................', this.state.popUpActiveItem);
```
Comment (izzett222) [inline src/components/dashboards/admin.js]: could you remove the console.log

File: src/components/popUp.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      textColor: '',
+    };
+    this.handleRoleChange = this.handleRoleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.setState({ email: '', role: '' });
+    await this.setState({ isPopUpActive: this.props.isPopUpActive });
+    console.log('..........DID MOUNT (popUp.js) .......props.......', this.props, '....state....', this.state);
```
Comment (izzett222) [inline src/components/popUp.js]: could you remove the console.log here too

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]

File: src/components/dashboards/admin.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      roles: ['manager', 'travel team member', 'requester', 'travel administrator', 'supplier', 'super administrator'],
+      popUp: 'invisible',
+      popUpActiveItem: {},
+    };
+    this.handleSetRolePopUp = this.handleSetRolePopUp.bind(this);
+    this.updateUsers = this.updateUsers.bind(this);
```
Comment (munyili-samuel) [inline src/components/dashboards/admin.js]: You do not need to bind arrow functions.
Ideally, you will use bind if the function is declared as `handleSetRolePopUp()`

File: src/components/dashboards/admin.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  async componentDidMount() {
+    const actionOutput = await this.props.getUsersAction();
+    this.setState({ users: actionOutput.payload.data });
+  }
+
+  handleSetRolePopUp = async (visibility, activeItem) => {
+    await this.setState({ popUp: visibility, popUpActiveItem: activeItem });
+  }
+
+  updateUsers = async (users) => {
```
Comment (munyili-samuel) [inline src/components/dashboards/admin.js]: No need to use `async` without `await`

File: src/scss/sidebar.scss [hunk 5]
```diff
@@ -7,7 +7,7 @@
   padding: 0 !important;
   z-index: 20;
   width: fit-content !important;
-  height: calc(100vh - 6.6rem);
+  height: calc(100vh - 0rem);
```
Comment (munyili-samuel) [inline src/scss/sidebar.scss]: Same as `100vh`

**Review by munyili-samuel — COMMENTED** [review]
You can ignore all the comments for now.
Just a way to share some knowledge

File: src/scss/sidebar.scss (hunk 5 above)
Comment (Baraka-Mugisha) [reply src/scss/sidebar.scss]: It has now been removed

File: src/components/dashboards/admin.js (hunk 3 above)
Comment (Baraka-Mugisha) [reply src/components/dashboards/admin.js]: I have corrected it

**Review by bbaime98 — APPROVED** [review]

**Pull request merged**